### PR TITLE
Migrate scaffold.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/scaffold.dart [hunk 1]
```diff
@@ -87,18 +86,18 @@ class DevToolsScaffold extends StatefulWidget {
...
   /// The page being rendered.
-  final String page;
+  final String? page;
 
   /// Whether to render the embedded view (without the header).
   final bool embed;
 
   /// IDE-supplied theming.
-  final IdeTheme ideTheme;
+  final IdeTheme? ideTheme;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/scaffold.dart:95]: done

File: packages/devtools_app/lib/src/shared/scaffold.dart [hunk 2]
```diff
@@ -114,16 +113,16 @@ class DevToolsScaffoldState extends State<DevToolsScaffold>
...
   /// This will be passed to both the [TabBar] and the [TabBarView] widgets to
   /// coordinate their animation when the tab selection changes.
-  TabController _tabController;
+  TabController? _tabController;
 
-  Screen _currentScreen;
+  late Screen _currentScreen;
 
-  ImportController _importController;
+  ImportController? _importController;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/scaffold.dart:120]: done

File: packages/devtools_app/lib/src/shared/scaffold.dart [hunk 3]
```diff
@@ -394,11 +393,11 @@ class DevToolsScaffoldState extends State<DevToolsScaffold>
 
     return Provider<BannerMessagesController>(
       create: (_) => BannerMessagesController(),
-      child: Provider<ImportController>.value(
+      child: Provider<ImportController?>.value(
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/scaffold.dart:396]: done

File: packages/devtools_app/lib/src/shared/scaffold.dart [hunk 4]
```diff
@@ -444,8 +445,8 @@ class DevToolsScaffoldState extends State<DevToolsScaffold>
   /// Builds an [AppBar] with the [TabBar] placed on the side or the bottom,
   /// depending on the screen width.
   Widget _buildAppBar(String title) {
-    Widget flexibleSpace;
-    Size preferredSize;
+    Widget? flexibleSpace;
+    late Size preferredSize;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/scaffold.dart:449]: the method will return early in this case. reformatted the code to make it clearer

File: packages/devtools_app/lib/src/shared/scaffold.dart [hunk 5]
```diff
@@ -583,7 +583,7 @@ class KeyboardShortcuts extends StatefulWidget {
 
 class KeyboardShortcutsState extends State<KeyboardShortcuts>
     with AutoDisposeMixin {
-  FocusNode _focusNode;
+  FocusNode? _focusNode;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/scaffold.dart]: done

File: packages/devtools_app/lib/src/shared/scaffold.dart [hunk 6]
```diff
@@ -117,10 +117,10 @@ class DevToolsScaffoldState extends State<DevToolsScaffold>
 
   late Screen _currentScreen;
 
-  ImportController? _importController;
+  late ImportController _importController;
 
-  StreamSubscription<ConnectVmEvent>? _connectVmSubscription;
-  StreamSubscription<String>? _showPageSubscription;
+  late StreamSubscription<ConnectVmEvent> _connectVmSubscription;
+  late StreamSubscription<String> _showPageSubscription;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/scaffold.dart:123]: can these be late final?

File: packages/devtools_app/lib/src/debugger/debugger_controller.dart [hunk 7]
```diff
@@ -4,6 +4,8 @@
 
 // @dart=2.9
 
+// ignore_for_file: avoid_redundant_argument_values
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/debugger/debugger_controller.dart:7]: My flutter version is equal to one in flutter-version.txt.
I get this warning for the files that are not migrated yet to 2.12. I plan to remove the annotations when I migrate the files.

File: packages/devtools_app/lib/src/shared/scaffold.dart (hunk 6 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/scaffold.dart]: done

**Comment by polina-c** [comment]
tested

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**